jax/_src/scipy/linalg.py

```diff
@@ -656,3 +656,13 @@ def sqrtm(A, blocksize=1):
   if blocksize > 1:
       raise NotImplementedError("Blocked version is not implemented yet.")
   return _sqrtm(A)
+
+@_wraps(scipy.linalg.hilbert)
+@partial(jit, static_argnames=("n",))
+def hilbert(n):
+  a = jnp.arange(n)
+  return _arange_to_hilbert(a)
+
+@jit
+def _arange_to_hilbert(a):
+  return 1 / (a[:, None] + a[None, :] + 1)
```

Comment:
As detailed in the comment in #10144, through this splitting of the function, the transformation part (`_arange_to_hilbert`) can be jitted completely, whereas `hilbert` must be decorated with `static_argnames`. If this is too verbose, I can change it. :)

Reply:
It's not necessary to split this into a separate helper function. Regarding JIT compilation, you've already decorated the main function wtih jit, so there is no benefit to separately jitting the implementation. Regarding code design, the two functions are so short that in my judgment delegating to a helper obscures rather than aids in understanding the code.

I'd suggest deleting lines 664-667